modules/claims_api/lib/auth_token_service/token_service.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+# require 'common/models/redis_store'
+
+module ClaimsApi
+  module AuthTokenService
+    class TokenService < ::Common::RedisStore
+      require_relative 'base'
+
+      redis_key :service_name
+
+      attribute :service_name, String
+      attribute :access_token, String
+
+      validates(:access_token, presence: true)
+
+      def self.access_token_exists?(service_name, tolerance)
+        return unless before_validations!(service_name)
+
+        set_redis_store(service_name)
+
+        token_exists = exists?(service_name)
+
+        if token_exists
+          access_token = find(service_name)
+          remaining_ttl = access_token.ttl
+
+          if remaining_ttl < tolerance
+            token_exists = false
+            access_token.destroy
+          end
+        end
+
+        token_exists
+      end
+
+      def self.set_access_token(service_name, access_token)
+        return unless before_validations!(service_name)
+
+        set_redis_store(service_name)
+        set_redis_ttl(service_name)
+
+        service = new(service_name:)
+        service.access_token = access_token
+        service.save!
+      end
+
+      def self.get_access_token(service_name)
+        return unless before_validations!(service_name)
+
+        service = find(service_name)
+
+        return nil if service.nil?
+
+        service&.access_token
+      end
```

Comment:
Did we peruse Jerek's changes? He added an AccessTokenTracker class we could probably utilize to cut this out. https://github.com/department-of-veterans-affairs/vets-api/blob/master/lib/lighthouse/auth/client_credentials/access_token_tracker.rb

Reply:
Ya I lifted some of this directly from that file.   I thought that because there were set values in there like this [this](https://github.com/department-of-veterans-affairs/vets-api/blob/master/lib/lighthouse/auth/client_credentials/access_token_tracker.rb#L6) we would not be able to utilize that file directly